src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,595 @@
+            resolutions (list or numpy array of shape num_filters containing integers): resolution of each filter function, ie number of intervals required to cover each filter image. If None, it is estimated from data.
+            gains (list or numpy array of shape num_filters containing doubles in [0,1]): gain of each filter function, ie overlap percentage of the intervals covering each filter image. If None, it is set as 0.33, since in the automatic parameter selection method in http://www.jmlr.org/papers/volume19/17-291/17-291.pdf, any arbitrary value between 1/3 and 1/2 works, so we go with the minimal one (ensuring that the complex is a graph if only given one filter).
+            clustering (class): clustering class (default sklearn.cluster.DBSCAN()). Common clustering classes can be found in the scikit-learn library (such as AgglomerativeClustering for instance). If None, it is set to hierarchical clustering, with scale estimated from data.
+            N (int): subsampling iterations (default 100) for estimating scale and resolutions. Used only if clustering or resolutions = None. See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            beta (double): exponent parameter (default 0.) for estimating scale and resolutions. Used only if clustering or resolutions = None. See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            C (double): constant parameter (default 10.) for estimating scale and resolutions. Used only if clustering or resolutions = None. See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            verbose (bool): whether to display info while computing.
+        """
+
+        self.filter_bnds, self.resolutions, self.gains, self.clustering = filter_bnds, resolutions, gains, clustering
+        self.input_type, self.mask, self.N, self.beta, self.C = input_type, min_points_per_node, N, beta, C
+        CoverComplexPy.__init__(self, verbose)
+
+    def estimate_scale(self, X, N=100, beta=0., C=10.):
+        """
+        Compute estimated scale of a point cloud or a distance matrix.
+
+        Parameters:
+            X (numpy array of shape (num_points) x (num_coordinates) if point cloud and (num_points) x (num_points) if distance matrix): input point cloud or distance matrix.
+            N (int): subsampling iterations (default 100). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            beta (double): exponent parameter (default 0.). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            C (double): constant parameter (default 10.). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+
+        Returns:
+            delta (double): estimated scale that can be used with eg agglomerative clustering.
```

Comment:
Sphinx turns this into
> Returns: estimated scale that can be used with eg agglomerative clustering.
> Return type: delta (double)

which seems wrong. Unless you know a way to let sphinx know about your syntax, we should change it to
```
            double: estimated scale that can be used with eg agglomerative clustering.
```
or
```
            double: delta: estimated scale that can be used with eg agglomerative clustering.
```

Reply:
Done!